src/flamegraph/svg.rs

```diff
@@ -48,224 +48,270 @@ pub(super) struct StyleOptions<'a> {
+    pub font_type: String,
+    pub font_size: usize,
+    // pub count_name: String,
+    pub name_type: String,
+    pub notes: String,
 }
 
-pub(super) fn write_prelude<'a, W>(
-    svg: &mut Writer<W>,
-    style_options: &StyleOptions<'a>,
-    opt: &Options,
-) -> quick_xml::Result<()>
-where
-    W: Write,
-{
-    svg.write_event(Event::Start(BytesStart::borrowed_name(b"defs")))?;
-    svg.write_event(Event::Start(BytesStart::borrowed(
-        br#"linearGradient id="background" y1="0" y2="1" x1="0" x2="0""#,
-        "linearGradient".len(),
-    )))?;
-    svg.write_event(Event::Empty(
-        BytesStart::borrowed_name(b"stop").with_attributes(
-            iter::once(("stop-color", &*style_options.bgcolor1))
-                .chain(iter::once(("offset", "5%"))),
-        ),
-    ))?;
-    svg.write_event(Event::Empty(
-        BytesStart::borrowed_name(b"stop").with_attributes(
-            iter::once(("stop-color", &*style_options.bgcolor2))
-                .chain(iter::once(("offset", "95%"))),
-        ),
-    ))?;
-    svg.write_event(Event::End(BytesEnd::borrowed(b"linearGradient")))?;
-    svg.write_event(Event::End(BytesEnd::borrowed(b"defs")))?;
+impl SVGWriter {
+    pub fn ypad2(self: &SVGWriter) -> usize {
+        self.font_size * 2 + 10
+    }
 
-    svg.write_event(Event::Start(
-        BytesStart::borrowed_name(b"style").with_attributes(iter::once(("type", "text/css"))),
-    ))?;
-    svg.write_event(Event::Text(BytesText::from_plain_str(
-        ".func_g:hover { stroke:black; stroke-width:0.5; cursor:pointer; }",
-    )))?;
-    svg.write_event(Event::End(BytesEnd::borrowed(b"style")))?;
+    pub fn write_header<W>(
+        self: &SVGWriter,
+        svg: &mut Writer<W>,
+        imageheight: usize,
+    ) -> quick_xml::Result<()>
+    where
+        W: Write,
+    {
+        svg.write(br#"<?xml version="1.0" standalone="no"?>"#)?;
+        svg.write(br#"<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">"#)?;
+        svg.write_event(Event::Start(
+            BytesStart::borrowed_name(b"svg").with_attributes(vec![
+                ("version", "1.1"),
+                ("width", &*format!("{}", self.image_width)),
+                ("height", &*format!("{}", imageheight)),
+                ("onload", "init(evt)"),
+                (
+                    "viewBox",
+                    &*format!("0 0 {} {}", self.image_width, imageheight),
+                ),
+                ("xmlns", "http://www.w3.org/2000/svg"),
+                ("xmlns:xlink", "http://www.w3.org/1999/xlink"),
+            ]),
+        ))?;
+        svg.write_event(Event::Comment(BytesText::from_plain_str(
+            "Flame graph stack visualization. \
+             See https://github.com/brendangregg/FlameGraph for latest version, \
+             and http://www.brendangregg.com/flamegraphs.html for examples.",
+        )))?;
+        svg.write_event(Event::Comment(BytesText::from_plain_str(
```

Comment:
Shouldn't this be only if notes isn't empty? I don't remember what the Perl code does.

Reply:
Perl code has default as empty string and use it in template without any condition.

https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L256